[1] AWS Compute Blog, “Running Simcenter STAR-CCM+ on AWS,” AWS Compute Blog, 2025, https://aws.amazon.com/blogs/compute/running-simcenter-star-ccm-on-aws/.

[2] Jules S. Damji, “Introduction to Ray for Distributed and Machine Learning Applications in Python,” YouTube, 2023, https://www.youtube.com/watch?v=w7uPnEqYz7A.

[3] Alejandro Fernandez‑Fraga, Jorge Gonzalez‑Domínguez, and María J. Martin, “Applying Dynamic Balancing to Improve the Performance of MPI Parallel Genomics Applications,” Proceedings of the 39th ACM/SIGAPP Symposium on Applied Computing, 2024, pp. 506–514, https://doi.org/10.1145/3605098.3635986.

[4] Katarína Grešová, Vlastimil Martinek, David Čechák, Petr Šimeček, and Panagiotis Alexiou, “Genomic benchmarks: a collection of datasets for genomic sequence classification,” BMC Genomic Data, vol. 24, 2023, p. 25, https://doi.org/10.1186/s12863-023-01123-8.

[5] Nagakishore Jammula, Sriram Chockalingam, and Srinivas Aluru, “Parallel Read Error Correction for Big Genomic Datasets,” Proceedings of the 22nd IEEE International Conference on High Performance Computing (HiPC 2015), 2015.

[6] Yanrong Ji, Zhihan Zhou, Han Liu, and Ramana V. Davuluri, “DNABERT: pre-trained Bidirectional Encoder Representations from Transformers model for DNA-language in genome,” Bioinformatics, vol. 37, no. 15, 2021, pp. 2112–2120, https://doi.org/10.1093/bioinformatics/btab083.

[7] Mani Khanuja, Farooq Sabir, Shreyas Subramanian, and Trenton Potgieter, “Applied Machine Learning and High-Performance Computing on AWS,” Packt Publishing, 2022, https://www.oreilly.com/library/view/applied-machine-learning/9781803237015/.

[8] John Koons, Florian Riedel, Surendra Shankar, and Hans Lutz, “Mashup: Making Serverless Computing Useful for HPC Workflows via Hybrid Execution,” Proceedings of the 26th ACM International Symposium on High-Performance Parallel and Distributed Computing (HPDC ’21), 2021, pp. 150–161, https://doi.org/10.1145/3503221.3508407.

[9] Wei Liu, Shu Chen, and Anil Kumar, “CodonBERT: Using BERT for Sentiment Analysis to Better Predict Genes with Low Expression,” 2023 IEEE International Conference on Computational Biology and Bioinformatics, 2023, pp. 78–86, https://doi.org/10.1145/3584371.3613013.

[10] Ion I. Măndoiu and Alexander Zelikovsky (eds.), “Computational Methods for Next Generation Sequencing Data Analysis,” Wiley Series on Bioinformatics: Computational Techniques and Engineering, John Wiley & Sons, 2016.

[11] Chandra Sekhar Mukhopadhyay, Ratan Kumar Choudhary, and Mir Asif Iquebal, “Basic Applied Bioinformatics,” John Wiley & Sons, 2018.

[12] Phi Nguyen and Vedant Jain, “Building Reliable and Scalable Generative AI Infrastructure on AWS with Ray and Anyscale,” AWS Partner Network Blog, 2023, https://aws.amazon.com/blogs/apn/building-reliable-and-scalable-generative-ai-infrastructure-on-aws-with-ray-and-anyscale/.

[13] Sung Park, Hyun Lee, Jihoon Kim, and Xiao Zhang, “A Survey of Processing Systems for Phylogenetics and Population Genetics,” Genetics and Molecular Biology, vol. 47, no. 2, 2024, pp. 123–145, https://doi.org/10.1145/3588033.

[14] Max Pumperla, Edward Oakes, and Richard Liaw, “Learning Ray: Flexible Distributed Python for Machine Learning,” O’Reilly Media, Inc., 2023, https://www.oreilly.com/library/view/learning-ray/9781098117214/.

[15] Abhishek Roy, Yanlei Diao, Uday Evani, Avinash Abhyankar, Clinton Howarth, Rémi Le Priol, and Toby Bloom, “Massively Parallel Processing of Whole Genome Sequence Data: An In-Depth Performance Study,” Proceedings of the 2017 ACM International Conference on Management of Data (SIGMOD ’17), 2017, pp. 187–202, https://doi.org/10.1145/3035918.3064048.

[16] Oguz Selvitopi et al., “Extreme-scale many-against-many protein similarity search,” Proceedings of the ACM SIGARCH International Conference on High Performance Computing, Networking, Storage and Analysis (SC ’22), 2022, https://doi.org/10.5555/3571885.3571887.

[17] David Spuler, “Tokenizer and Vocabulary,” in *Generative AI in C++*, Chapter 27, Aussie AI, 2024, https://www.aussieai.com/book/ch27-tokenizer-embedding.

[18] Mark Vinciguerra, Anoop Saha, Florian Stahl, and Alex Iankoulski, “Ray jobs on Amazon SageMaker HyperPod: scalable and resilient distributed AI,” AWS Machine Learning Blog, 2025, https://aws.amazon.com/blogs/machine-learning/ray-jobs-on-amazon-sagemaker-hyperpod-scalable-and-resilient-distributed-ai/.

[19] Feng Zhang, Mei Xu, and Kai Lin, “LSHvec: A Vector Representation of DNA Sequences Using Locality Sensitive Hashing and FastText Word Embeddings,” Bioinformatics, vol. 37, no. 21, 2021, pp. 3890–3897, https://doi.org/10.5555/3459930.3469521.

[20] Yu Zhou, Ming Yang, and Kevin Thomson, “MoTeX: A Word-Based HPC Tool for MoTif eXtraction,” Proceedings of the 1st ACM International Workshop on Genomics and Big Data (GenBD ’13), 2013, pp. 12–19, https://doi.org/10.1145/2506583.2506587.

[21] Greg Wilson, Sarah Owen, Paul Hargrove, and Allen Malony, “Experiences with Performing MapReduce Analysis of Scientific Data on HPC Platforms,” Proceedings of the 2016 IEEE International Parallel and Distributed Processing Symposium Workshops, 2016, pp. 1053–1060, [https://doi.org/10.1145/2912152.2912154](https://doi.org/10.1145/2912152.2912154).

[22] Awni Hannun, Jagrit Digani, Angelos Katharopoulos, and Ronan Collobert, “MLX: Efficient and flexible machine learning on Apple silicon,” version 0.0, GitHub repository, 2023, [https://github.com/ml-explore](https://github.com/ml-explore) (installation guide: [https://ml-explore.github.io/mlx/build/html/install.html](https://ml-explore.github.io/mlx/build/html/install.html)).

[23] Nikolay A. Simakov, Matthew D. Jones, Thomas R. Furlani, Eva Siegmann, and Robert J. Harrison. 2024. First Impressions of the NVIDIA Grace CPU Superchip and NVIDIA Grace Hopper Superchip for Scientific Workloads. In International Conference on High Performance Computing in Asia-Pacific Region Workshops (HPCAsiaWS 2024), January 25–27, 2024, Nagoya, Japan. ACM, New York, NY, USA, 9 pages. https://doi.org/10.1145/3636480.3637097